searchguard/rolesmapping.py

```diff
@@ -0,0 +1,157 @@
+            raise DeleteRoleMappingException('Error deleting the role mapping for role {} '
+                                             '- msg: {}'.format(role, delete_sg_rolemapping.text))
+    else:
+        # Raise exception because the role mapping does not exist
+        raise DeleteRoleMappingException('Error deleting the role mapping for role {}, does not exist'.format(role))
+
+
+def create_rolemapping(role, properties):
+    """Creates a Search Guard role mapping. Returns when successfully created
+    It is required to specify at least one of: users, backendroles or hosts in the properties argument.
+    We do not use the PATCH endpoint for backwards compatibility with Elasticsearch before 6.4.0
+
+    :param str role: Name of the role mapping to create in Search Guard
+    :param dict properties: Search Guard role mapping fields (users, backendroles and/or hosts)
+    :raises: CreateRoleMappingException, CheckRoleExistsException
+    """
+    if not check_role_exists(role):
+        raise CheckRoleExistsException('Role {} does not exist'.format(role))
+
+    properties_keys = {"users", "backendroles", "hosts"}
+
+    if not any(key in properties for key in properties_keys):
+        # Raise exception because we did not receive valid properties
+        raise CreateRoleMappingException('Error creating mapping for role {} - Include at least one of: users, '
+                                         'backendroles or hosts keys in the properties argument'.format(role))
+
+    __process_data(role, properties)
+    return
+
+
+def modify_rolemapping(role, properties, action="replace"):
+    """Modifies a Search Guard role mapping. Returns when successfully modified
+    It is required to specify at least one of: users, backendroles or hosts in the properties argument.
+    We do not use the PATCH endpoint for backwards compatibility with Elasticsearch before 6.4.0
+
+    :param str role: Name of the role mapping to create in Search Guard
+    :param dict properties: Search Guard role mapping fields (users, backendroles and/or hosts)
+    :param boolean action: Defines what to do with the properties. Defaults to replace (overwrites existing
+    properties). Other options are merge (combine the properties with existing ones) or split
+    (removes the properties from existing ones)
+    :raises: ModifyRoleMappingException
+    """
+    if not check_rolemapping_exists(role):
+        raise ModifyRoleMappingException('Mapping for role {} does not exist'.format(role))
+
+    properties_keys = {"users", "backendroles", "hosts"}
+
+    if not any(key in properties for key in properties_keys):
+        # Raise exception because we did not receive valid properties
+        raise ModifyRoleMappingException('Error modifying mapping for role {} - Include at least one of: users, '
+                                         'backendroles or hosts keys in the properties argument'.format(role))
```

Comment:
Maybe this entire `if` could be put in a separate method to keep things a little more DRY.